internal/datanode/flow_graph_dd_node_test.go

```diff
@@ -12,19 +12,231 @@
 package datanode
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/milvus-io/milvus/internal/msgstream"
+	"github.com/milvus-io/milvus/internal/proto/commonpb"
+	"github.com/milvus-io/milvus/internal/proto/datapb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+	"github.com/milvus-io/milvus/internal/util/flowgraph"
 )
 
-func TestFlowGraphDDNode_Operate(t *testing.T) {
-	// clearSignal := make(chan UniqueID)
-	// collectionID := UniqueID(1)
-	// vchanInfo := &datapb.VchannelInfo{
-	//     CollectionID: collectionID,
-	// }
-	// ddNode := newDDNode(clearSignal)
-	// ddNode := newDDNode()
+func TestFlowGraph_DDNode(te *testing.T) {
+	te.Run("Test DDNode newDDNode", func(t *testing.T) {
+		tests := []struct {
+			inCollID UniqueID
+
+			inFlushedSegs        []UniqueID
+			inUnFlushedSegID     UniqueID
+			inUnFlushedChannelTs Timestamp
+		}{
+			{UniqueID(1), []UniqueID{100, 101, 102}, 200, 666666},
+			{UniqueID(2), []UniqueID{103}, 200, 666666},
+			{UniqueID(3), []UniqueID{}, 200, 666666},
+			{UniqueID(3), []UniqueID{104}, 0, 0},
+		}
+
+		for _, test := range tests {
+			di := &datapb.SegmentInfo{}
+
+			if test.inUnFlushedSegID != 0 {
+				di.ID = test.inUnFlushedSegID
+				di.DmlPosition = &internalpb.MsgPosition{Timestamp: test.inUnFlushedChannelTs}
+			}
+
+			ddNode := newDDNode(
+				make(chan UniqueID),
+				test.inCollID,
+				&datapb.VchannelInfo{
+					FlushedSegments:   test.inFlushedSegs,
+					UnflushedSegments: []*datapb.SegmentInfo{di},
+				},
+			)
+
+			assert.Equal(t, "ddNode", ddNode.Name())
+			assert.Equal(t, test.inCollID, ddNode.collectionID)
+			assert.Equal(t, len(test.inFlushedSegs), len(ddNode.flushedSegments))
```

Comment:
We can add a description field to test case and put these into a t.Run(), so we can find which failed case quickly.